Articles/Blog/WhenIsBetterToWriteAutomatedTests-src.md

```diff
@@ -0,0 +1,112 @@
+
+## Notes
+
+If the first four criteria above are satisfied but it is not too hard to write automated tests, then often one should write them anyway because it will make the code easier extend and to maintain going forward.
+
+Adjectives like "minor", "obvious", "easy", and "hard" are all subjective and do not have precise definitions.
+For example, while for one person might consider these five criteria as "minor", "obvious", "easy", "easy" and "hard", another person may consider them "significant", "non-obvious", "not easy", "painful", and/or "tractable".
+For example, if one does not know the sensing, separation, and fake collaborators strategies for unit testing described in *Working Effectively with Legacy Code*, then one might think that adding automated tests for a piece of software is "hard" while another more skilled, knowledgeable, and/or experienced developer might consider adding tests for that piece of software to be quite tractable.
+
+## Summary
+
+While it often pays of to write a high quality automated test suite for a piece of software (i.e. reduce initial development costs and improve long-term maintenance), there are situations where it does not pay off.  Here we listed five criteria that if satisfied, it is better not to write automated tests and instead do manual testing when any changes are made to the code.
+
+## Disclaimer
+
+Please don't use this blog article as an excuse for not writing automated tests for some piece of software by trying to convince yourself that the damage of defects will be "minor", any failures will be "obvious", or the defects will be "easy" for anyone to fix.
+You might think that but your users, stakeholders, and other developers (who will need to maintain this software) may not feel the same way.
+So please use discretion when applying this criteria to some piece of software when deciding whether or not to write automated tests (and how much testing is reasonable).
+
+<br>
+
+### Author bio
+
+[Roscoe A. Bartlett](https://bartlettroscoe.github.io) is a computational scientist and engineer in the [Center for Computing Reserach](https://cfwebprod.sandia.gov/cfdocs/CompResearch/templates/insert/research.cfm) at [Sandia National Laboratories](https://sandia.gov).
+Roscoe is a long-time contributor to the [Trilinos Project](https://trilinos.org) and is currently working on various subprojects in the [Exascale Computing Project](https://www.exascaleproject.org/) including [IDEAS-ECP](https://ideas-productivity.org/ideas-ecp/).
+
+[1]: https://bssw.io/items/working-effectively-with-legacy-code/ "Feathers, Micheal C. Working Effectively with Legacy Code.  Prentice Hall, 2004, ISBN: 0131177052 {}"
+[2]: https://doi.org/10.1109/eScience.2012.6404448/ "Overview of the TriBITS lifecycle model: A Lean/Agile software lifecycle model for research-based computational science and engineering software {}"
```

Comment:
The link here goes to an error page. When I removed the trailing / it worked fine.

Reply:
Thinks for catching this.  Fixed in https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/pull/377/commits/117bf705ee4ff211f347fc04bb70f89b93baa8c8#diff-ff08f44603f4e10f3095baf6cea0da6eR97.

See:

* https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/8946fafd601a7e1d05bf00769c4a091e7e906974/Articles/Blog/WhenIsBetterToWriteAutomatedTests.md#ref2